docs/intro/overrides.rst

```diff
@@ -0,0 +1,387 @@
+            ...  # more specific parsing
+
+    @handle_urls(["dualexample.com", "dualexample.net"], overrides=GenericProductPage)
+    class DualExampleProductPage(ItemWebPage):
+        def to_item(self):
+            ...  # more specific parsing
+
+The code above declares that:
+
+    - For sites that matches the ``example.com`` pattern, ``ExampleProductPage``
+      would be used instead of ``GenericProductPage``.
+    - The same is true for ``YetAnotherExampleProductPage`` where it is used
+      instead of ``GenericProductPage`` for two URLs: ``dualexample.com`` and
+      ``dualexample.net``.
+    - However, ``AnotherExampleProductPage`` is only used instead of ``GenericProductPage``
+      when we're parsing pages from ``anotherexample.com`` which doesn't contain
+      ``/digital-goods/`` in its URL path.
+
+The override mechanism that ``web-poet`` offers could still be further
+customized. You can read some of the specific parameters and alternative ways
+to organize the rules via the :ref:`Overrides API section <api-overrides>`.
+
+To demonstrate another alternative way to declare the Override rules, see the
+code example below:
+
+.. code-block:: python
+
+    from web_poet.pages import ItemWebPage
+    from web_poet import PageObjectRegistry
+
+    primary_registry = PageObjectRegistry()
+    secondary_registry = PageObjectRegistry()
+
+    class GenericProductPage(ItemWebPage):
+        def to_item(self):
+            return {"product title": self.css("title::text").get()}
+
+    @primary_registry.handle_urls("example.com", overrides=GenericProductPage)
+    class ExampleProductPage(ItemWebPage):
+        def to_item(self):
+            ...  # more specific parsing
+
+    @secondary_registry.handle_urls("anotherexample.com", overrides=GenericProductPage, exclude="/digital-goods/")
+    class AnotherExampleProductPage(ItemWebPage):
+        def to_item(self):
+            ...  # more specific parsing
+
+    @primary_registry.handle_urls(["dualexample.com", "dualexample.net"], overrides=GenericProductPage)
+    @secondary_registry.handle_urls(["dualexample.com", "dualexample.net"], overrides=GenericProductPage)
+    class DualExampleProductPage(ItemWebPage):
+        def to_item(self):
+
+If you need more control over the Registry, you could instantiate your very
+own :class:`~.PageObjectRegistry` and use its ``@handle_urls`` to annotate and
+register the rules. This might benefit you in certain project use cases where you
+need more organizational control over your rules.
+
+Viewing all available Overrides
+-------------------------------
+
+A convenience function is available discover and retrieve all rules from your
+project. Make sure to check out :ref:`Overrides API section <api-overrides>`
+to see the other functionalities.
+
+.. code-block:: python
+
+    from web_poet import default_registry
+
+    # Retrieves all rules that were registered in the registry
+    rules = default_registry.get_overrides()
```

Comment:
If I'm not mistaken, this example has a big caveat: all modules which use handle_urls decorator must be imported already, before calling `default_registry.get_overrides()`.

Reply:
I.e. if you don't call osme kind of walk_modules, or have some framework doing it at a right time, there is a risk ruless are going to be missing.

Reply:
Gotcha. I see what you mean now. I've created a new `consume_modules()` function that can be _optionally_ called before `default_registry.get_overrides()`, should the need arises. Also updated the docs to properly reflect this caveat.

Updated this on: https://github.com/scrapinghub/web-poet/pull/16/commits/daa3ff9bdfa754bacfd8a20862f68acb12a3fa61